File: lessons/lesson2.py

# this is called a class
class Square:
    def __init__(self):
        print('I am a square')

# this is another class
class Rectangle:
    def __init__(self, attitude):
        print('I am a ' + attitude + ' circle')

if __name__ == '__main__':
    # create a square object
    s = Square()

    # create a rectangle, and pass in an argument
    r = Rectangle('funny')

File: lessons/lesson2.cpp
#include<iostream>


// this is called a class
class Square
{
public:
    Square()
    {
        std::cout << "I am a square" << std::endl;
    }
};
// this is another class
class Rectangle
{
public:
    Rectangle(auto attitude)
    {
        std::cout << "I am a "<< " " <<  attitude<< " " <<  " circle" << std::endl;
    }
};
;
int main()
{
    // create a square object
    Square s;
    // create a rectangle, and pass in an argument
    Rectangle r("funny");
}